From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Thu, 19 Aug 2021 17:27:01 +0300
Subject: [PATCH 29/67] Do not create CreateFallbackSurface. JB#55226
 OMP#JOLLA-296

Fallback surface creation was introduces in gecko mirror
sha1 369ac65ffe5ef (https://bugzilla.mozilla.org/show_bug.cgi?id=1453501)

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 gfx/gl/GLContextProviderEGL.cpp | 27 +--------------------------
 1 file changed, 1 insertion(+), 26 deletions(-)

diff --git a/gfx/gl/GLContextProviderEGL.cpp b/gfx/gl/GLContextProviderEGL.cpp
index cfe28b1f9517..8931365ebee4 100644
--- a/gfx/gl/GLContextProviderEGL.cpp
+++ b/gfx/gl/GLContextProviderEGL.cpp
@@ -144,31 +144,6 @@ static bool is_power_of_two(int v) {
   return (v & (v - 1)) == 0;
 }
 
-static EGLSurface CreateFallbackSurface(EglDisplay& egl,
-                                        const EGLConfig& config) {
-  if (egl.IsExtensionSupported(EGLExtension::KHR_surfaceless_context)) {
-    // We don't need a PBuffer surface in this case
-    return EGL_NO_SURFACE;
-  }
-
-  std::vector<EGLint> pbattrs;
-  pbattrs.push_back(LOCAL_EGL_WIDTH);
-  pbattrs.push_back(1);
-  pbattrs.push_back(LOCAL_EGL_HEIGHT);
-  pbattrs.push_back(1);
-
-  for (const auto& cur : kTerminationAttribs) {
-    pbattrs.push_back(cur);
-  }
-
-  EGLSurface surface = egl.fCreatePbufferSurface(config, pbattrs.data());
-  if (!surface) {
-    MOZ_CRASH("Failed to create fallback EGLSurface");
-  }
-
-  return surface;
-}
-
 static EGLSurface CreateSurfaceFromNativeWindow(
     EglDisplay& egl, const EGLNativeWindowType window, const EGLConfig config) {
   MOZ_ASSERT(window);
@@ -348,7 +323,7 @@ GLContextEGL::GLContextEGL(const std::shared_ptr<EglDisplay> egl,
       mSurfaceConfig(surfaceConfig),
       mContext(context),
       mSurface(surface),
-      mFallbackSurface(CreateFallbackSurface(*mEgl, mSurfaceConfig)) {
+      mFallbackSurface(EGL_NO_SURFACE) {
 #ifdef DEBUG
   printf_stderr("Initializing context %p surface %p on display %p\n", mContext,
                 mSurface, mEgl->mDisplay);
